### PR TITLE
add CONTRIBUTING.md and Table of Contents

### PR DESCRIPTION
**Pull request by hakilebara**
* Move contribution guidelines away from README.md to CONTRIBUTING.md.
* Add a ToC to README.md (generated with https://github.com/jonschlinkert/markdown-toc)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -378,27 +398,10 @@ export default Ember.Route.extend({
 });
 ```
 
-# Installation
```
Comment (Alonski) [inline README.md]: I am not sure that removing this section from the README should be done.
Most Ember Addons have this as part of the README and I feel that it is expected of Addons to have this section.

@bendemboski @knownasilya What do you think?

File: README.md [hunk 2]
```diff
@@ -4,6 +4,26 @@
 
 To use the uploader, you must provide a name (for proper queueing and bundling of resources), and an upload URL.
 
+## Table of Contents
```
Comment (Alonski) [inline README.md]: Nice!

File: README.md (hunk 1 above)
Comment (hakilebara) [reply README.md]: Contribution information for Ember Addons is now split by default https://github.com/ember-cli/ember-cli/pull/8203

More context:
https://noti.st/kennethlarsen/jXt2dj/the-state-of-community-documentation#sE2Y4Nq

Comment (knownasilya) [reply README.md]: This is a different installation section. This is for the consumer not the contributor.

**Comment by Alonski** [comment]
Thanks @hakilebara !

**Pull request merged**